src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs

```diff
@@ -290,7 +293,8 @@ internal SqlStatistics Statistics
             {
                 if (null != _activeConnection)
                 {
-                    if (_activeConnection.StatisticsEnabled)
+                    if (_activeConnection.StatisticsEnabled || 
+                        _diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
```

Comment:
In the tests there is a toy observer called `FakeDiagnosticListenerObserver `. In the tests it gets hooked up to the SqlClient DiagnosticListener on line `DiagnosticListener.AllListeners.Subscribe(diagnosticListenerObserver);` (where `diagnosticListenerObserver` is a `FakeDiagnosticListenerObserver`).